signac/core/zarrcollection.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Implements Zarr-backend.
+
+This implements the Zarr-backend for SyncedCollection API by
+implementing sync and load methods.
+"""
+from .synced_collection import SyncedCollection
+from .syncedattrdict import SyncedAttrDict
+from .synced_list import SyncedList
+
+
+class ZarrCollection(SyncedCollection):
+    """Implement sync and load using a Zarr backend."""
+
+    backend = __name__  # type: ignore
+
+    def __init__(self, name=None, store=None, **kwargs):
+        import zarr
+        import numcodecs  # zarr depends on numcodecs
+
+        self._root = zarr.group(store=store)
```

Comment:
Similar to the MongoDB and Redis backends, I suggest a slight shift of the API here. Since we are not using the `store` argument anywhere else, it would probably make sense to directly pass the `root` group to the constructor.